service/history/workflow/mutable_state_impl.go

```diff
@@ -2912,14 +2912,17 @@ func (ms *MutableStateImpl) ApplyActivityTaskScheduledEvent(
 		}
 	}
 
+	ms.AddPendingActivityInfo(ai)
+	ms.writeEventToCache(event)
+	return ai, nil
+}
+
+func (ms *MutableStateImpl) AddPendingActivityInfo(ai *persistencespb.ActivityInfo) {
```

Comment:
it is used in tests.

Reply:
I agree. But `ai.ScheduledEventId` is a key in the map. If it is missing then it will be `0` and I guess many things will break. @ychebotarev, if you plan to use it somewhere else you need to add some validation here.

Reply:
removed, I will reuse ApplyScheduledEvent